ipt.sh

```diff
@@ -48,126 +54,168 @@ else
-    -m string --algo bm --hex-string '|ffffffff54|' \
-    -j DROP
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
-    -m string --algo bm --hex-string '|ffffffff54|' \
-    -j LOG ${LOGLIMIT} --log-ip-options \
-    --log-prefix "${LOGPREFIX} a2s flood: "
+# create our ipset rules - requires ipset!
+ipset create permatrusted hash:ip    timeout 10800 || true
+ipset create signedon     hash:ip    timeout 10800 || true
 
-## 6: UDP spam (25 req/s limit)
-## We should never be seeing 25 packets a second from the same ip to the same port not already established or related
-RULE_FILTER="-m hashlimit --hashlimit-name speedlimit --hashlimit-mode srcip,dstport --hashlimit-above 25/sec"
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
-    -j DROP
+#########################################################################
+# PREROUTING RAW
+#########################################################################
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
-    -j LOG ${LOGLIMIT} --log-ip-options \
-    --log-prefix "${LOGPREFIX} >25 req/s: "
 
-
-## 5: PACKET TOO smol: There should never be any packets packets below 32 bytes.
+# 7: PACKET TOO SMOL
+# There should never be any packets packets below 32 bytes
 RULE_FILTER="-m length --length 0:32"
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT} ${ports} ${RULE_FILTER}    \
     -j DROP
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
-    -j LOG ${LOGLIMIT} --log-ip-options \
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT} ${ports} ${RULE_FILTER}    \
+    -j LOG ${LOGLIMIT} --log-ip-options                                     \
     --log-prefix "${LOGPREFIX} len < 32: "
 
-## 4: PACKET TOO BIG: There should never be any packets above the following length.
-## (net_maxroutable) + (net_splitrate) * (net_maxfragments)
-##  1260             +  1              *  1260
-##  = 2521 bytes
+
+# 6: PACKET TOO BIG
+# There should never be any packets above this length:
+# (net_maxroutable) + (net_splitrate  * net_maxfragments)
+#  1260             + (1              *  1260)
+#  = 2520 (+1) bytes
 RULE_FILTER="-m length --length 2521:65535"
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT} ${ports} ${RULE_FILTER}    \
     -j DROP
 
-${ipt} -p udp ${COMMENT} ${ports} ${RULE_FILTER} \
-    -j LOG ${LOGLIMIT} --log-ip-options \
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT} ${ports} ${RULE_FILTER}    \
+    -j LOG ${LOGLIMIT} --log-ip-options                                     \
     --log-prefix "${LOGPREFIX} len > 2521: "
 
+RULE_FILTER="-m string --algo bm --hex-string"
 
-## 3: UDP game packets
-## Allow "Established" packets so that we dont stomp on legit gamers
-## This rule goes last so it gets inserted first
-RULE_FILTER="-m state --state ESTABLISHED,RELATED"
+# 5: Log client signons
+${ipt_pre_raw} -p udp -i ${defaultin} ${COMMENT}                            \
```

Comment:
A lot of duplicate code here. Consider turning into a function and calling it in a loop.